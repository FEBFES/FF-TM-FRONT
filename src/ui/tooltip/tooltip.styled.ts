import styled, { css, RuleSet } from 'styled-components';
import { PlacementType } from './tooltip';

export const SRelativeContainer = styled.div`
  position: relative;
`;

export type StyledVariants<E extends string> = {
  [key in E]?: RuleSet;
};

const tooltipPlacements: StyledVariants<PlacementType> = {
  left: css`
    top: 50%;
    transform: translateY(-50%);
    right: 120%;
  `,
  top: css`
    bottom: 120%;
    left: 50%;
    transform: translateX(-50%);
  `,
  right: css`
    top: 50%;
    transform: translateY(-50%);
    left: 120%;
  `,
  bottom: css`
    top: 120%;
    left: 50%;
    transform: translateX(-50%);
  `,
};

export const STooltip = styled.div<{ placement: PlacementType }>`
  ${({ placement }) => tooltipPlacements[placement]};
  font-size: 12px;
  color: var(--font-defautl);
  min-width: fit-content;
  position: absolute;
  background: var(--bg-0);
  border: 1px solid var(--bg-border);
  padding: 4px;
  border-radius: 4px;
`;
