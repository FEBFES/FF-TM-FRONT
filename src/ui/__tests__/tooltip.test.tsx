import React from 'react';
import { Tooltip } from '../tooltip/tooltip';
import renderer from 'react-test-renderer';

it('SNAPSHOTS: UI - Tooltip component - render', () => {
  const elem = renderer
    .create(
      <Tooltip title="test" placement="left">
        <div>asd</div>
      </Tooltip>
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});
