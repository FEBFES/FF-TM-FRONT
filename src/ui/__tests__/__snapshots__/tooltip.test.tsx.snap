// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SNAPSHOTS: UI - Tooltip component - render 1`] = `
<div
  className="sc-bcPKhP dqyMgV"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="sc-bcPKhP dqyMgV"
  >
    <div>
      asd
    </div>
  </div>
</div>
`;
